src/integrationTest/java/uk/gov/hmcts/reform/bulkscanprocessor/helpers/DirZipper.java

```diff
@@ -0,0 +1,43 @@
+package uk.gov.hmcts.reform.bulkscanprocessor.helpers;
+
+import com.google.common.io.Files;
+
+import java.io.ByteArrayOutputStream;
+import java.io.File;
+import java.io.IOException;
+import java.util.List;
+import java.util.stream.Stream;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipOutputStream;
+
+import static com.google.common.io.Resources.getResource;
+import static java.util.stream.Collectors.toList;
+
+public class DirZipper {
+
+    /**
+     * Zips files from given directory. Files in resulting archive are NOT wrapped in a directory.
+     */
+    public static byte[] zipDir(String dirName) throws IOException {
+
+        ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+        try (ZipOutputStream zos = new ZipOutputStream(outputStream)) {
+
+            // .listFiles() does not guarantee any order.
+            // Files need to be sorted as currently mocks rely on files order.
+            // TODO: improve mocks in tests so that order does not matter
+            List<File> filesToZip =
+                Stream.of(new File(getResource(dirName).getPath()).listFiles())
+                    .sorted()
+                    .collect(toList());
```

Comment:
could you just do a forEach here or is the order not guaranteed?

Reply:
can't zip operations throw a checked exception 😢

Reply:
try catch / wrapper function?

Reply:
gets too bloated imho